subsys/usb/class/hid/core.c

```diff
@@ -127,6 +127,10 @@ struct hid_device_info {
 #ifdef CONFIG_USB_HID_BOOT_PROTOCOL
 	uint8_t protocol;
 #endif
+	enum usb_dc_status_code status;
```

Comment:
Maybe we could keep the `usb_dc_status_code` (except suspend/resume/sof) and `bool suspended` flag?
This could prevent inconsistency that could occur for current implementation (we do it in similar way for nrf_desktop)

Reply:
I removed `usb_dc_status_code` and added `bool suspended` as You proposed. This is done similar as #27025. This PR is fixing an issue. I the near future I wish to implement per USB device state tracing and the whole problem with flags here will be a history.